pkg/testcase/selinux.go

```diff
@@ -0,0 +1,246 @@
+func TestSelinuxContext() {
+	cluster := factory.AddCluster(GinkgoT())
+	product, err := shared.GetProduct()
+	if err != nil {
+		log.Println(err)
+	}
+
+	if cluster.NumServers > 0 {
+		for _, ip := range cluster.ServerIPs {
+			var context map[string]string
+			context, err := getContext(product, ip)
+			Expect(err).NotTo(HaveOccurred())
+
+			fmt.Print("\nThese are the whole commands to use in this context validation\n")
+			fmt.Print("Command to run || Context expected\n")
+			for cmdsToRun, contExpected := range context {
+				fmt.Println(cmdsToRun + " || " + contExpected)
+			}
+
+			for cmd, expectedContext := range context {
+				res, err := shared.RunCommandOnNode(cmd, ip)
+				fmt.Println("\nRunning cmd:", cmd, "\nExpected context:", expectedContext)
+				fmt.Println("Result: \n", res)
+				if res != "" {
+					Expect(res).Should(ContainSubstring(expectedContext), "Error on cmd %v \n Context %v \nnot found on ", cmd, expectedContext, res)
+					Expect(err).NotTo(HaveOccurred())
+				}
+			}
+		}
+	}
+}
+
+func getVersion(osRelease, ip string) string {
+	if strings.Contains(osRelease, "VERSION_ID") {
+		res, err := shared.RunCommandOnNode("cat /etc/os-release | grep 'VERSION_ID'", ip)
+		Expect(err).NotTo(HaveOccurred())
+		parts := strings.Split(res, "=")
+		if len(parts) == 2 {
+			// Get version
+			version := strings.Trim(parts[1], "\"")
+			// if dot exist get the first number
+			if dotIndex := strings.Index(version, "."); dotIndex != -1 {
+				version = version[:dotIndex]
+			}
+			return version
+		}
+	}
+	return ""
+}
+
+var osPolicy string
+
+func getContext(product, ip string) (cmdCtx, error) {
+	res, err := shared.RunCommandOnNode("cat /etc/os-release", ip)
+	if err != nil {
+		return nil, err
+	}
+
+	fmt.Println("OS Release: \n", res)
+	policyMapping := map[string]string{
+		"ID_LIKE='suse' VARIANT_ID='sle-micro'": "sle_micro",
+		"ID_LIKE='suse'":                        "micro_os",
+		"ID_LIKE='coreos'":                      "coreos",
+		"VARIANT_ID='coreos'":                   "coreos",
+	}
+
+	for k, v := range policyMapping {
+		if strings.Contains(res, k) {
+			return selectPolicy(product, v), nil
+		}
+	}
+
+	version := getVersion(res, ip)
+	versionMapping := map[string]string{
+		"7": "centos7",
+		"8": "centos8",
+		"9": "centos9",
+	}
+
+	if policy, ok := versionMapping[version]; ok {
+		return selectPolicy(product, policy), nil
+	}
+
+	return nil, fmt.Errorf("unable to determine policy for %s on os: %s", ip, res)
+}
+
+func selectPolicy(product, osType string) cmdCtx {
+	key := fmt.Sprintf("%s_%s", product, osType)
+
+	for _, config := range conf {
+		if config.distroName == key {
+			fmt.Printf("\nUsing '%s' policy for this %s cluster.\n", osType, product)
+			osPolicy = osType
+			return config.cmdCtx
+		}
+	}
+
+	fmt.Printf("Configuration for %s not found!\n", key)
+	return nil
+}
+
+// TestSelinuxSpcT Validate that containers don't run with spc_t
+func TestSelinuxSpcT() {
```

Comment:
this should be renamed to - fetchauxZProcess  ( or something similar ) 
and should come from shared.aux
and here u just use your value ( spc_t )

Reply:
As we discussed in a call, we'll leave this as it is